aten/src/ATen/native/TensorShape.cpp

```diff
@@ -340,9 +362,6 @@ std::vector<Tensor> split_with_sizes(const Tensor& self, IntList split_sizes, in
          << "entries, but got split_sizes=" << split_sizes;
       throw std::runtime_error(ss.str());
     }
-    if (start_idx >= dim_size) {
-      break;
-    }
```

Comment:
Is this the behavior change we talked about?

Reply:
I think you are referring to "include the index to the end" discussion, but no, I decided not to do that because IMO we should just have a separate function that behaves like numpy split.

This change just prevents us from exiting out of the loop early (the previous assumption was that each size was at least 1, so we knew we'd be past the dim_size the next time through the loop, but that isn't correct either because we had size [0] tensors).

In any case, narrow already does the correct things in all these cases (error checking, etc.), so there's no need for this code.